### DISCUSSION TITLE
Traffic watch QueryTimings < ElapsedMilliseconds

### DISCUSSION THREAD

**Discussion by JoshuaLevett** [ravendb/ravendb#19565] [Q&A] [answered]
I have several traffic watch logs that look like the below.
What would cause such a difference between the ElapsedMilliseconds and the QueryTimings breakdown?
`{
    "TrafficWatchType": "Http",
    "TimeStamp": "2024-11-05T14:21:37.0463213Z",
    "DatabaseName": "production",
    "CustomInfo": "from index 'index' where bool = $p0 and string = $p1 select id() as Id limit $p2, $p3\n{\"p0\":true,\"p1\":\"string\",\"p2\":0,\"p3\":1}",
    "ClientIP": "...",
    "CertificateThumbprint": "...",
    "RequestId": 43619,
    "HttpMethod": "POST",
    "ElapsedMilliseconds": 3018,
    "ResponseStatusCode": 200,
    "RequestUri": "...",
    "AbsoluteUri": "...",
    "RequestSizeInBytes": 195,
    "ResponseSizeInBytes": 1112,
    "Type": "Queries",
    "QueryTimings": {
      "DurationInMs": 3,
      "Timings": {
        "Query": {
          "DurationInMs": 3,
          "Timings": {
            "Lucene": {
              "DurationInMs": 3,
              "Timings": null
            },
            "Retriever": {
              "DurationInMs": 0,
              "Timings": {
                "Projection": {
                  "DurationInMs": 0,
                  "Timings": {
                    "Storage": {
                      "DurationInMs": 0,
                      "Timings": null
                    }
                  }
                }
              }
            }
          }
        },
        "Staleness": {
          "DurationInMs": 0,
          "Timings": null
        }
      }
    }
  }`

**Comment by arekpalinski** [comment, marked as answer]
Hi @JoshuaLevett,

`ElapsedMilliseconds` applies to a whole request duration (reading the request body, handling / execution of a request, writing a response to a network stream), while `QueryTimings` is just about the timings of a query execution by a search engine.

**Reply by JoshuaLevett** [reply]
Thanks @arekpalinski 
Is there any way I can dig into where that time (over 3 seconds) is being spent? This doesn't have a particularly large request or response.

Could it be throttled by I/O or network bandwidth?

Considering I have another sample with an ElapsedMilliseconds of 118 and a QueryTimings of 105 (and it has both a bigger request and response) it is a bit of a mystery to me.
We are with RavenDB Cloud, so this might make sense as a support ticket now

**Reply by arekpalinski** [reply]
There are multiple factors that might contribute to this, so yes we'd need to check internal RavenDB Cloud monitoring of your instance. There is number of possibilities depending on load and resources utilization that you have there.
